docs/adr/2023-08-18_025-resource-based-api.md

```diff
@@ -0,0 +1,146 @@
+    binary (`cbor`) output of `transaction` fields in several of these using a
+    query parameter.
+
+- Many of these features have been added in a "quick and dirty" way, by monkey
+  patching the encoded JSON.
+
+- The current capabalities even do not satisfy all user needs:
+
+  - Need to wade through lots of events to know the latest state (except the
+    very basic `headStatus` from the `Greetings`).
+
+  - Need to poll `GetUTxO` _or_ aggregate confirmed transactions on client side
+    to know the latest UTxO for constructing transactions.
+
+  - Inclusion of the whole UTxO in the head is not always desirable and
+    filtering by address would be beneficial. (not addressed in this ADR though)
+
+  - As [ADR-15](/adr/15) also proposes, some clients may not need (or should
+    not have) access to administrative information.
+
+- It is often a good idea to separate the responsibilities of Commands and
+  Queries (CQRS), as well as the model they use.
+
+## Decision
+
+- Do not add a general purpose querying interface (e.g. graphql) to the
+  `hydra-node` and stay true to [ADR-3](/adr/3) of using duplex communication
+  channels (using websockets).
+
+- **Drop** `GetUTxO` and `GetUTxOResponse` messages as they advocate a
+  request/response way of querying.
+
+- Realize that `ClientInput` data is actually a `ClientCommand` (renaming them)
+  and that `ServerOutput` are just `projections` of the [internal event stream
+  (see ADR-24)](/adr/24) into read `models` on the API layer.
+
+- Compose the API out of resource `models`, which compartmentalize the domain
+  into topics on the API layer.
+
+  - Each resource has a _latest_ state, which can be queried or clients may
+    subscribe to any model changes.
+
+  - A resource's `model` type needs to be a result of a pure `projection` from
+    server output events, i.e. `project :: model -> ServerOutput -> model`.
+
+    - TBD: or use the event stream type `StateChanged` instead? potentially
+      renaming it?
+
+  - Each resource is available at some HTTP path, also called "endpoint":
+
+    - `GET` requests must respond with the _latest_ state in a single response.
+
+    - `GET` requests with websocket upgrade headers must start a websocket
+      connection, push the _latest_ state as first message and any resource
+      state updates after.
+
+    - Other HTTP verbs may be accepted by a resource handler, i.e. to issue
+      resource-specific commands. Any commands accepted must also be available
+      via the corresponding websocket connection.
+
+  - Each resource is only available as `JSON` encoding (for now).
+
+    - (TBD): already specify that `Accept` headers should be used to negotiate a
+      `content-type`?
+
+- (TBD): Keep the direct/raw output of `ServerOutput` events on `/`, which
+  also accepts all `ClientCommand` messages.
+
+### Resources
+
+Resource paths + HTTP verbs mapped to existing things:
```

Comment:
We should already define a versioning scheme, either as part of the path or as a content-type

Reply:
I added a `/v1` instead of the `/node`